src/System.Linq/src/System/Linq/AppendPrepend.cs

```diff
@@ -243,6 +245,25 @@ public override List<TSource> ToList()
                 return list;
             }
 
+            public override HashSet<TSource> ToHashSet(IEqualityComparer<TSource> comparer)
+            {
+                HashSet<TSource> hashSet = new HashSet<TSource>(comparer);
+
+                if (!_appending)
+                {
+                    hashSet.Add(_item);
+                }
+
+                hashSet.UnionWith(_source);
+
+                if (_appending)
```

Comment:
Ok, but we need pass the `_source`, right?

`HashSet<TSource> hashSet = new HashSet<TSource>(_source, comparer); hashSet.Add(_item);`